### ISSUE TITLE
Septentrio twos-complement error?

### ISSUE THREAD

**Issue by rwpenney** [tomojitakasu/RTKLIB#270]
The decoding of twos-complement values within the Septentrio File Format (SBF) reader (https://github.com/tomojitakasu/RTKLIB/blob/master/src/rcv/septentrio.c, near line-467) does not appear to be correct for fields that are not exactly eight bits wide.

In particular, the MeasEpoch blocks (used for GPS/GLONASS/Galileo multi-frequency observations) uses some 3-bit and 5-bit twos-complement fields to represent pseudo-range and Doppler offsets between different frequency channels. The code within septentrio.c currently applies a conversion routine compTwoConv() that only seems to handle 8-bit fields correctly. For example, in a 3-bit field, it is an unsigned value greater than 3, rather than 127 that indicates that the decoded (signed) value is negative. This is likely to mean that the decoding of L2 carrier-phase values are incorrect.

**Comment by JensReimann** [comment]
You are right. The problem has already been fixed (among many others) in my tree (https://github.com/JensReimann/RTKLIB/tree/rtklib_2.4.3), but not yet merged to Tomoji's branch. 
If you are interessed in Septentrio support please try this code.
